pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -99,8 +103,41 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
       return false;
     }
 
-    Comparable value = getValue(filterQueryTree.getValue().get(0), metadata.getDataType());
     PartitionFunction partitionFunction = metadata.getPartitionFunction();
+
+    if (operator == FilterOperator.EQUALITY) {
+      return pruneEquality(filterQueryTree, metadata, partitionRanges, partitionFunction);
+    }
+
+    return pruneIn(filterQueryTree, metadata, partitionRanges, partitionFunction);
```

Comment:
We have already returned if it is not one of  EQUALITY or IN operator earlier. I will add another test just to be clearer.